Calculate scores for qualitative data upon submit.

tomaszgy opened the pull request:
cf. [OC-4383](https://tasks.opencraft.com/browse/OC-4383)

This PR adds processing of the qualitative data and stores output of group membership calculation as a set of Score records. It also adds extra tests for LDA model.

**Test instructions:**
- Run tests

**Reviewers:**
- [x] @itsjeyd 

Signed-off-by: Tomasz Gargas <tomasz@opencraft.com>

itsjeyd commented on README.md:
```diff
@@ -19,9 +19,13 @@ Setting up the development server
+Please provide such a model and a vectorizer,
+replacing `lda.pkl` and `tfidf_vectorizer.pkl` files
+in the root directory of the project with your files.
+(Currently, `lda.pkl` and `tfidf_vectorizer.pkl` are the empty Python pickle files.)
```
@tomaszgy Typo: Replace "the" with "just":

> ... are **just** empty Python pickle files.

tomaszgy replied:
👍

itsjeyd commented on lpd/models.py:
```diff
@@ -154,6 +154,9 @@ def update_scores(cls, learner):
         based on all of the learner's QualitativeAnswers.
+
+        Scores need to be equal to (1 - probability) to have
+        the desired effect on recommendations generated by the adaptive engine.
```
@tomaszgy Very nice, thanks for adding this 😀

tomaszgy replied:
😃

itsjeyd commented on lpd/qualitative_data_analysis.py:
```diff
@@ -85,7 +67,7 @@ def calculate_probabilities(answers):
     document = make_document(answers)
     tfidf_matrix = make_tfidf_matrix([document])
-    weights = settings.LDA_MODEL.transform(tfidf_matrix)
+    weights = settings.LDA_MODEL.transform(tfidf_matrix)[0, :]
```
@tomaszgy Why do we need to slice here? (Just asking because I'd like to understand, not suggesting this needs to be changed.)

tomaszgy replied:
@itsjeyd `transform` returns a matrix with a single row. So in order to have a list of weights (the single row mentioned before), we need to use the above Numpy's syntax.

itsjeyd commented on lpd/tests/test_qualitative_data_analysis.py:
```diff
@@ -51,3 +54,49 @@ def test_make_document(self, answers, expected_document):
+                'kc_id_4_set-me': 0.03020937,
+                'kc_id_5_set-me': 0.03021411,
+                'kc_id_6_set-me': 0.81875085,
+                'kc_id_7_set-me': 0.03019817,
```
@tomaszgy The `_set-me` suffix is kind of confusing in this context, I think we can remove it :)

tomaszgy replied:
@itsjeyd Good catch :) Just removed the suffix.

itsjeyd commented on lpd/tests/test_qualitative_data_analysis.py:
```diff
@@ -51,3 +54,49 @@ def test_make_document(self, answers, expected_document):
+            'kc_id_7_set-me'
+        ]
+    )
+    def test_calculate_probabilities(self, answers, expected_probabiities):
```
@tomaszgy Could you add a docstring to this test?

tomaszgy replied:
👍

tomaszgy commented:
@itsjeyd I've addressed your comments and updated the description of this PR.

Regarding `tfidf_vectorizer.pkl`, that's an empty pickle file so I think it's fine to keep it here. We will need to provide proper `tfidf_vectorizer.pkl` on production.

Note that there's also a `lpd/tests/tfidf_vectorizer_test.pkl` file - it was created for the testing purposes similarly to `lpd/tests/lda_test.pkl`.

itsjeyd commented on lpd/tests/test_qualitative_data_analysis.py:
```diff
@@ -90,13 +90,24 @@ def test_make_document(self, answers, expected_document):
+        """
         probabilities = qda.calculate_probabilities(answers)
+        self.assertEqual(
+            expected_probabiities.keys(),
```
@tomaszgy Ack, sorry for not noticing this in the test's signature earlier, but there's a typo here: `probabiities` (missing `l`).

tomaszgy replied:
@itsjeyd No worries and good catch :) Just corrected.

itsjeyd commented:
For future reference, we discussed the issue mentioned [above](https://github.com/open-craft/learner-profile-dashboard/pull/7#pullrequestreview-119806373) in [this thread](https://chat.opencraft.com/opencraft/pl/rz4k3awe7pfnpncy1mjykpuc7a) on Mattermost.

itsjeyd commented:
@tomaszgy I think I accidentally overwrote the changes that you made to include test instructions in the PR description :/ Sorry about that! If you have a copy of the test instructions somewhere, could you paste them back in?

tomaszgy commented:
@itsjeyd No worries, I wrote it here again ;)

itsjeyd commented:
@tomaszgy Great, thanks a lot!
